pg_search/src/api/index.rs

```diff
@@ -595,3 +599,98 @@ pub fn term_set(
 
     SearchQueryInput::TermSet { terms }
 }
+
+#[pg_extern]
+pub fn dump_bm25(
+    index_name: String,
+) -> TableIterator<'static, (name!(heap_tid, i64), name!(content, pgrx::JsonB))> {
+    let bm25_index_name = format!("{}_bm25_index", index_name);
+
+    let database_oid = crate::MyDatabaseId();
+    let index_oid = index_oid_from_index_name(&bm25_index_name);
```

Comment:
we'll need to hold an `AccessShareLock` on the index for the duration of this function to make sure it's not dropped out from underneath us.

Once you've gotten the index_oid, you can do this:

```rust
// hold an AccessShareLock on the index relation to prevent concurrent index drops
let _index_relation = PgRelation::with_lock(index_oid, pg_sys::AccessShareLock as _);
```

And that'll hold the lock until the function ends.